test/test_nn.py

```diff
@@ -3282,6 +3282,54 @@ def forward(self, X):
             parametrize.type_before_parametrizations(model) == original_type
         )
 
+    def test_deepcopy_after_parametrization(self):
+        r"""Test that we are able to create a deepcopy of the module when it's parametrized."""
+
+        class AddOne(nn.Module):
+            def forward(self, x):
+                return x + 1.0
+
+        class ModelWithoutDeepcopy(nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.weight = nn.Parameter(torch.tensor([1., 1., 1., 1.]), requires_grad=True)
+                self.bias = nn.Parameter(torch.tensor([0., 0., 0., 0.]), requires_grad=True)
+                self.attr = [1.0, 2.0, 3.0, 4.0]
+
+        class ActualModel(ModelWithoutDeepcopy):
+            # Emulate custom implementation of the deepcopying.
```

Comment:
I don't think this is a very accurate emulation. For testing purposes, you can do a dummy function here that just returns a new Module.

Reply:
Yes, it's not an accurate deepcopy, but if I create a new Module like `self.__class__()` I should register all parametrizations once again for this module depending on ones that have already registered for `self`. 
As this simple module contains all weights, biases and attributes in `__dict__` don't you think that it's enough just to create a `deepcopy` of it's `__dict__`?

Reply:
Ok, We can keep it then.